### PR TITLE
Delete uneeded source functionalities and clean-up

### PR DESCRIPTION
**Pull request by chraibi**
While generating the positions of the sources, there was
some hacking and trying to adapt their velocity (vector) to their
environment.

This often necessitates a call to the router (so that the desired direction can be
initialized).

However, since the positions are not set yet, the router may choke on these agents
without a meaningful position (defaults to (0,0)).

In this PR the following happens wrt. sources:

- Unused functions are deleted
- Positions of sources are initialized
- *Afterwards*...
- adjust the velocity of the source-agents by making a call to
- `AdjustVelocityUsingWeidmann()`. Within this function, there is a unique call to
  the router.
- Refactoring a for-loop using function `GenerateAgents()`

TODO:

- `AgentsSourcesManager::AdjustVelocityUsingWeidmann()` is not tested in this PR
- Alternatively the function `AgentsSourcesManager::AdjustVelocityByNeighbour()`
  can be used as well.

Related to #303

### CODE REVIEW

File: libcore/src/pedestrian/AgentsSource.cpp [hunk 1]
```diff
@@ -73,7 +73,7 @@ AgentsSource::~AgentsSource() {}
 void AgentsSource::GenerateAgentsAndAddToPool(int count, Building * building)
 {
     std::vector<Pedestrian *> peds;
-    GenerateAgents(peds, count, building);
+    GenerateAgents(peds, count, building); // positions are not set yet!
```
Comment (schroedtert) [inline libcore/src/pedestrian/AgentsSource.cpp]: Could you add this comment as comment for GenerateAgents in the header file?

File: libcore/src/pedestrian/AgentsSourcesManager.cpp [hunk 2]
```diff
@@ -140,6 +133,10 @@ bool AgentsSourcesManager::ProcessAllSources() const
             } else if(!ComputeBestPositionVoronoiBoost(src.get(), peds, _building, source_peds))
                 LOG_WARNING("There was no place for some pedestrians");
 
+            // Having set the positions, now we can set the velocity
+            for(auto ped : peds)
+                AdjustVelocityUsingWeidmann(ped);
+
```
Comment (schroedtert) [inline libcore/src/pedestrian/AgentsSourcesManager.cpp]: Please add braces:
```
for(auto ped : peds){
    AdjustVelocityUsingWeidmann(ped); 
}
```

File: libcore/src/pedestrian/AgentsSource.cpp [hunk 3]
```diff
@@ -220,27 +220,20 @@ const std::shared_ptr<StartDistribution> AgentsSource::GetStartDistribution() co
...
+    //          the positions will be calculated and initializes.
+    // TODO: We should reverse the order of points 1 and 2.
     int pid;
     pid = (this->GetAgentId() >= 0) ?
               this->GetAgentId() :
               Pedestrian::GetAgentsCreated() + building->GetAllPedestrians().size();
     for(int i = 0; i < count; i++) {
         if(GetStartDistribution()) {
             auto ped = GetStartDistribution()->GenerateAgent(building, &pid, emptyPositions);
-            if(ped->FindRoute() == -1) {
```
Comment (schroedtert) [inline libcore/src/pedestrian/AgentsSource.cpp]: `FindRoute` is now only called after the position is set?

File: libcore/src/pedestrian/AgentsSourcesManager.cpp [hunk 4]
```diff
@@ -53,12 +53,7 @@ void AgentsSourcesManager::Run()
     LOG_INFO("Starting agent manager thread");
     //Generate all agents required for the complete simulation
     //It might be more efficient to generate at each frequency step
-    //TODO  this loop is exactly GenerateAgents( --> REFACTOR)
-    for(const auto & src : _sources) {
-        LOG_INFO("Generate src: {}", src->GetId());
-        src->GenerateAgentsAndAddToPool(src->GetMaxAgents(), _building);
-    }
-
+    GenerateAgents();
```
Comment (schroedtert) [inline libcore/src/pedestrian/AgentsSourcesManager.cpp]: Are all pedestrian computed beforehand or only the ones added in this step? Is this comment still valid?

File: libcore/src/pedestrian/AgentsSource.cpp (hunk 1 above)
Comment (chraibi) [reply libcore/src/pedestrian/AgentsSource.cpp]: ok

File: libcore/src/pedestrian/AgentsSourcesManager.cpp (hunk 2 above)
Comment (chraibi) [reply libcore/src/pedestrian/AgentsSourcesManager.cpp]: ok

File: libcore/src/pedestrian/AgentsSource.cpp (hunk 3 above)
Comment (chraibi) [reply libcore/src/pedestrian/AgentsSource.cpp]: Yes, in `AdjustVelocityUsingWeidmann(ped)` called in `AgentsSourcesManager.cpp`

File: libcore/src/pedestrian/AgentsSourcesManager.cpp (hunk 4 above)
Comment (chraibi) [reply libcore/src/pedestrian/AgentsSourcesManager.cpp]: This is just refactoring. Functionality was not altered. 
Here, for every source a `MaxAgents` are generated and added to Pool, waiting for `processAllSources` to be added to `building`.

**Pull request merged**